monkey/monkey_island/cc/ui/src/components/types/MapNode.tsx

```diff
@@ -0,0 +1,12 @@
+export class MapNode {
+    machine_id!: number
+    operating_system!: string
+    hostname!: string
+    network_interfaces!: object
+    agent_is_running!: boolean
+    island!: boolean
+    propagated_to!: boolean
+    connections!: object
```

Comment:
What's the point of `!`? If the field is required, let's enforce that it's passed during the construction. For that we need `constructor(machine_id, operating_system="unknown", hostname="", network_interfaces....){ this.machine_id = machine_id; ... }`